tests/log_parser/test_utils.py

```diff
@@ -0,0 +1,117 @@
+import os
+from copy import deepcopy
+
+import pytest
+import simplejson as json
+from django.db.utils import DataError
+from jsonschema import ValidationError
+
+from treeherder.log_parser.utils import (_lookup_extra_options_max,
+                                         validate_perf_data)
+
+
+@pytest.fixture
+def length_ok():
+    return {
+        'framework': {},
+        'suites': [
+            {
+                'extraOptions': [
+                    '.' * 45,
+                    '.' * 100,
+                ],
+                'name': 'testing',
+                'subtests': []
+            }
+        ] * 3
+    }
+
+
+@pytest.fixture
+def longer_than_all_max():
+    return {
+        'framework': {},
+        'suites': [
+            {
+                'extraOptions': [
+                    '.' * 46,
+                    '.' * 101,
+                ],
+                'name': 'testing',
+                'subtests': []
+            }
+        ]
+    }
+
+
+@pytest.fixture
+def longer_than_max():
+    return {
+        'framework': {},
+        'suites': [
+            {
+                'extraOptions': [
+                    '.' * 45,
+                    '.' * 101,
+                ],
+                'name': 'testing',
+                'subtests': []
+            }
+        ]
+    }
+
+
+@pytest.fixture
+def longer_than_second_max():
+    return {
+        'framework': {},
+        'suites': [
+            {
+                'extraOptions': [
+                    '.' * 46,
+                    '.' * 100,
+                ],
+                'name': 'testing',
+                'subtests': []
+            }
+        ] * 3
+    }
+
+
+@pytest.fixture
+def wrong_data(longer_than_all_max, longer_than_max, longer_than_second_max):
+    return deepcopy((longer_than_all_max, longer_than_max, longer_than_second_max))
+
+
+def test_extra_option_max_length():
+    with open(os.path.join('schemas', 'performance-artifact.json')) as f:
+        PERFHERDER_SCHEMA = json.load(f)
+    assert 100 == _lookup_extra_options_max(PERFHERDER_SCHEMA)
+
+
+def test_validate_perf_schema_no_exception(length_ok):
+    try:
+        validate_perf_data(length_ok)
+    except ValidationError as exc:
+        pytest.fail(str(exc))
+
+
+def test_validate_perf_schema(wrong_data):
```

Comment:
If we use global vars, then we can parameterize this function.

Reply:
OK